wpilibNewCommands/src/main/java/edu/wpi/first/wpilibj2/command/ParallelDeadlineGroup.java

```diff
@@ -81,9 +81,26 @@ public final void addCommands(Command... commands) {
     CommandScheduler.getInstance().registerComposedCommands(commands);
 
     for (Command command : commands) {
-      if (!Collections.disjoint(command.getRequirements(), getRequirements())) {
+      var sharedRequirements = new HashSet<>(this.getRequirements());
+      sharedRequirements.retainAll(command.getRequirements());
+      if (!sharedRequirements.isEmpty()) {
+        StringBuilder sharedRequirementsStr = new StringBuilder();
+        boolean first = true;
+        for (Subsystem requirement: sharedRequirements) {
+          if (first) {
+            first = false;
+          } else {
+            sharedRequirementsStr.append(", ");
+          }
+          sharedRequirementsStr.append(requirement.getName());
+        }
         throw new IllegalArgumentException(
-            "Multiple commands in a parallel group cannot require the same subsystems");
+          String.format(
+            "Command %s could not be added to this ParallelCommandGroup"
```

Comment:
I suggest avoiding the word `ParallelCommandGroup` as that class is only one of the ones relevant. "Parallel composition" would be more accurate.